rflx/generator/session.py

```diff
@@ -0,0 +1,2796 @@
+    ) -> Sequence[Statement]:
+        return [
+            # WORKAROUND: Componolit/Workarounds#32
+            PragmaStatement(
+                "Warnings",
+                [
+                    Variable("Off"),
+                    String(f'unused assignment to "{element_context}"'),
+                ],
+            ),
+            PragmaStatement(
+                "Warnings",
+                [
+                    Variable("Off"),
+                    String(f'"{element_context}" is set by "Update" but not used after the call'),
+                ],
+            ),
+            CallStatement(
+                sequence_type * "Update",
+                [
+                    Variable(sequence_context),
+                    Variable(element_context),
+                ],
+            ),
+            PragmaStatement(
+                "Warnings",
+                [
+                    Variable("On"),
+                    String(f'"{element_context}" is set by "Update" but not used after the call'),
+                ],
+            ),
+            PragmaStatement(
+                "Warnings",
+                [
+                    Variable("On"),
+                    String(f'unused assignment to "{element_context}"'),
+                ],
+            ),
+        ]
+
+    @staticmethod
+    def _allocate_buffer(identifier: rid.ID, initialization: Expr = None) -> Assignment:
+        return Assignment(
+            buffer_id(identifier),
+            New(
+                QualifiedExpr(
+                    const.TYPES_BYTES,
+                    initialization
+                    if initialization
+                    else NamedAggregate(
+                        (
+                            ValueRange(
+                                First(const.TYPES_INDEX),
+                                Add(First(const.TYPES_INDEX), Number(4095)),
+                            ),
+                            First(const.TYPES_BYTE),
+                        )
+                    ),
+                )
+            ),
+        )
+
+    @staticmethod
+    def _initialize_context(
+        identifier: rid.ID, type_: ID, first: Expr = None, last: Expr = None
+    ) -> CallStatement:
+        return CallStatement(
+            type_ * "Initialize",
+            [
+                Variable(context_id(identifier)),
+                Variable(buffer_id(identifier)),
+                *(
+                    [
+                        first
+                        or Call(const.TYPES_TO_FIRST_BIT_INDEX, [First(buffer_id(identifier))]),
+                        last or Call(const.TYPES_TO_FIRST_BIT_INDEX, [Last(buffer_id(identifier))]),
+                    ]
+                    if first or last
+                    else []
+                ),
+            ],
+        )
+
+    def _copy_to_buffer(self, type_: ID, source_context: ID, target_buffer: ID) -> IfStatement:
+        """Might raises a deferred exception."""
```

Comment:
raises -> raise

Reply:
Changed in b436967812ba47e1c6c93f77f456b9b9676ff3c3.